examples/nlp/character_and_byte_level_text_generation.py

```diff
@@ -0,0 +1,285 @@
+model_unicode.compile(loss="sparse_categorical_crossentropy", optimizer=optimizer)
+
+x_byte = keras.Input(shape=(None,))
+e_byte = tf.keras.layers.Embedding(input_dim = byte_vocab_size, output_dim = 128)(x_byte)
+y_byte = keras.layers.LSTM(128)(e_byte)
+z_byte = keras.layers.Dense(byte_vocab_size, activation="softmax")(y_byte)
+model_byte = keras.Model(x_byte, z_byte)
+
+optimizer = keras.optimizers.RMSprop(learning_rate=0.01)
+model_byte.compile(loss="sparse_categorical_crossentropy", optimizer=optimizer)
+
+"""
+Now we'll train both our models for 6 Epochs post which the loss starts increasing once again
+"""
+
+model_unicode.fit(unicode_encoded_dataset, epochs=6)
+model_byte.fit(byte_encoded_dataset, epochs=6)
+
+"""
+We'll now showcase the `keras_nlp.utils.greedy_search` utility offered by KerasNLP which essentially outputs the 
+most likely token at each time step!
+
+We'll generate outputs for the same prompt from both the models!
+
+You might notice the .decode() call on some strings this is because the tensorflow tensors uses byte strings internally 
+which can't directly be interpreted as things such as accents are replaced by special symbols and need to be decoded 
+to a utf-8 format to be interpreted well
+"""
+
+def decode_sequences(input_sentences, model, generation_length = 50):
+    tokenized = UnicodeCharacterTokenizer(input_sentences)
+    prompt = tokenized.to_tensor(shape=tokenized.shape.as_list()[:-1] + [tf.strings.length(input_sentences[0])])
+    def token_probability_fn(input):
+        if (model == 'model_unicode'):
+          x = model_unicode(input)[-1, :]
+        elif (model == 'model_byte'):
+          x = model_byte(input)[-1, :]
+        return tf.expand_dims(x, axis=0)
+    generated_tokens = keras_nlp.utils.greedy_search(
+        token_probability_fn,
+        prompt = prompt,
+        max_length=100+generation_length,
+        )
+    generated_sentences = UnicodeCharacterTokenizer.detokenize(generated_tokens)
+    return generated_sentences[0]
+
+for i in range(5):
+  start_index = random.randint(0, len(text) - 100 - 1)
+  sentence = text[start_index: start_index + 100]
+  models = ['model_unicode', 'model_byte']
+  print(f"EXAMPLE {i+1}")
+  print("PROMPT: ", sentence)
+  print()
+  for model in models:
+    translated = decode_sequences(input_sentences = tf.constant([sentence]), model = model, generation_length = 50)
+    print(f"Model: {model}")
+    print("OUTPUT: ",translated.numpy().decode())
+    print()
+
+"""
+EXAMPLE 1
+PROMPT:   que ya su dulce esposo no vivía, rompió los aires con suspiros, hirió los cielos con quejas, maltra
+
+Model: model_unicode
+OUTPUT:   que ya su dulce esposo no vivía, rompió los aires con suspiros, hirió los cielos con quejas, maltrae. Ɏ y el como son prés como sancho de la carón
+
+Model: model_byte
+OUTPUT:   que ya su dulce esposo no vivía, rompió los aires con suspiros, hirió los cielos con quejas, maltra de su don quijoteâ, que su perreso de su cab
```

Comment:
We need to keep looking at where these erroneous characters are coming from `â`, maybe we need a longer dataset? Different hyperparameters? Maybe trying the transfomer will help?

Especially with greed search, we should at least be able to train to the point where we output a valid set of characters that are actually in the source text.